modules/reitit-ring/src/reitit/ring.cljc

```diff
@@ -48,6 +67,46 @@
           (respond (error-handler request)))
         (respond (not-found request)))))))
 
+#?(:clj
+   (defn create-resource-handler
+     "A ring handler for serving classpath resources, configured via options:
+
+     | key          | description |
+     | -------------|-------------|
+     | :parameter   | optional name of the wildcard parameter, defaults to unnamed keyword `:`
+     | :root        | optional resource root, defaults to `\"public\"`
+     | :mime-types  | optional extension->mime-type mapping, defaults to `reitit.ring.mime/default-types`
+     | :path        | optional path to mount the handler to. Works only if mounted outside of a router."
+     ([]
+      (create-resource-handler nil))
+     ([{:keys [parameter root mime-types path]
+        :or {parameter (keyword "")
+             root "public"
+             mime-types mime/default-mime-types}}]
+      (let [response (fn [file]
+                       {:status 200
+                        :body file
+                        :headers {"Content-Type" (mime/ext-mime-type (.getName file) mime-types)}})]
+        (if path
+          (let [path-size (count path)]
+            (fn
+              ([req]
+               (let [uri (:uri req)]
+                 (if (and (>= (count uri) path-size))
+                   (some->> (str root (subs uri path-size)) io/resource io/file response))))
+              ([req respond _]
+               (let [uri (:uri req)]
```

Comment:
Could simplify by naming this function with `(fn foo ...)` and then here `(respond (foo req))`.